app/models/service_template_ansible_tower.rb

```diff
@@ -24,7 +24,7 @@ def self.create_catalog_item(options, _auth_user = nil)
 
   def remove_invalid_resource
     # remove the resource from both memory and table
-    service_resources.to_a.delete_if { |r| r.destroy unless r.resource(true) }
+    service_resources.to_a.delete_if { |r| r.destroy unless r.resource.reload }
```

Comment:
So I think you have this backwards... Or, at least I think you do based on what _**I think**_ this is a better link for the "change commit" (compared to what you have in the description):

https://github.com/rails/rails/commit/6eae366d0d2e5d5211eeaf955f56bd1dc6836758

Since this is a singular association, I think you want `r.reload.resource` to clear the objects "association cache", as it were, and then load the object again and confirm nothing is returned from the DB.

The downside I see with this is that it could be more SQL queries then what you have here (one to reload the `r`, and one to reload the `r.resource` again), but I think this is then in the spirit of what was trying to be done here.  Possibly there is another internal API to just clear the association cache, but unsure.  Also unsure if a `reload` will mess with the `r.destroy`.

Reply:
This is why I asked you @NickLaMuro.  I think you're absolutely right.  

While you raise good questions about the number of sql queries, etc., I don't think this is ever called often enough to cause performance concern.

Reply:
@jrafanie that is where I would ask the original author about this, since I am just guessing what the code is doing based on context (jedi mind trick).  Couldn't tell you which is supposed to be "more right".